components/SubgraphEditor/SubgraphConfig/SelectedContract/EventRow.tsx

```diff
@@ -109,22 +111,20 @@ export const EventRow = (props: EventRowProps) => {
           .reduce((acc: any[], { options }) => [...acc, ...options], [])
           .find(mfs => mfs.value === eventHandler.handler)}
         name="handler"
-        onChange={handleSelectOptionChange('handler')}
+        onChange={newVal => {
+          if (newVal!.label == newFnName) {
+            createMappingFn(newFnName, eventName)
+          }
+          handleUpdate({ ...eventHandler, handler: newVal!.label })
+        }}
         options={mappingFnsOptions}
         isLoading={fnExtractionLoading}
         formatGroupLabel={formatGroupLabel}
         placeholder="Declare event handler function"
       />
-      <ActionBtnsContainer editing={eventHandler.editing}>
-        {eventHandler.editing ? (
-          <Check size={16} color="#ffffff" onClick={saveEventHandler} />
-        ) : (
-          <>
-            <Search size={16} />
-            <Edit size={16} onClick={toggleEditing} />
```

Comment:
This was done by design, to make it more conscious of the editing process. Plus it's a good safeguard against unfinished edits. The user might select an event, but then leave out the event handler. That can cause errors.